src/iomusxml.cpp

```diff
@@ -1639,6 +1640,34 @@ void MusicXmlInput::ReadMusicXmlDirection(
         }
     }
 
+    // Principal voice
+    pugi::xpath_node lead = type.node().select_node("principal-voice");
+    if (lead) {
+        int voiceNumber = wedge.node().attribute("number").as_int();
+        voiceNumber = (voiceNumber < 1) ? 1 : voiceNumber;
+        if (HasAttributeWithValue(lead.node(), "type", "stop")) {
+            std::vector<std::pair<BracketSpan *, musicxml::OpenSpanner> >::iterator iter;
+            for (iter = m_bracketStack.begin(); iter != m_bracketStack.end(); ++iter) {
+                    int measureDifference = m_measureCounts.at(measure) - iter->second.m_lastMeasureCount;
+                    iter->first->SetTstamp2(std::pair<int, double>(measureDifference, timeStamp));
+                    m_bracketStack.erase(iter);
+                    return;
+            }
```

Comment:
There is a problem with this loop. It makes no sense since it contains a `return` statement

Reply:
Yes, sorry. The original plan was to allow nested brackets. But those shouldn't turn up, so I took the shortcut. 
Want to do? Remove the loop entirely or add an only an if statement for later improvements?

Reply:
Please remove the loop. (Double check that this does not create a memory leak with objects remaining in the m_bracketStack)